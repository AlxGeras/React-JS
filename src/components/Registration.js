import { TextField, Button } from "@mui/material";
import { Link } from "react-router-dom";
import { useRef, useState } from "react"
import { AssignmentInd, Login } from "@mui/icons-material";
import firebaseConfig from "../services/firebaseConfig";
import { createUserWithEmailAndPassword, getAuth } from "firebase/auth";
import { ToastContainer, toast } from 'react-toastify'
import FormHelperText from '@mui/material/FormHelperText';
import { useInputStyles } from "../styles/inputStyles";
import { useDispatch } from "react-redux";
import { addProfileWithFB } from "../midllewares/middleware";


const Registration = () => {
    const dispatch = useDispatch()

    const [email, setEmail] = useState('');
    const [password, setPassword] = useState('');
    const [name, setName] = useState('');
    const [error] = useState('');

    const handleEmailChange = (e) => {
        setEmail(e.target.value)
    }

    const handlePasswordChange = (e) => {
        setPassword(e.target.value)
    }

    const handleNameChange = (e) => {
        setName(e.target.value)
    }

    const onSubmit = async (e) => {
        e.preventDefault();
        try {
            const auth = getAuth(firebaseConfig);
            await createUserWithEmailAndPassword(auth, email, password);
            dispatch(addProfileWithFB(name, email))
            toast.success('Пользователь успешно зарегестрирован');
            setEmail('');
            setPassword('');

        } catch (e) {
            console.log(e);
        }
    }

    const classes = useInputStyles();
    const inputField = useRef();


    return <div className={classes.formWrap}>
        <ToastContainer />

        <form className={classes.form} onSubmit={onSubmit}>
            <fieldset className={classes.inputScreen}>
                <legend className={classes.legend}>
                    Форма регистрации
                </legend>
                <TextField
                    inputRef={inputField}
                    variant="outlined"
                    size="small"
                    type='text'
                    label="Введите ваше имя"
                    className={classes.inputLine}
                    value={name}
                    helperText={" "}
                    onChange={handleNameChange}
                    autoFocus
                    required
                />
                <TextField
                    inputRef={inputField}
                    variant="outlined"
                    size="small"
                    type='email'
                    label="Введите email"
                    className={classes.inputLine}
                    value={email}
                    helperText={" "}
                    onChange={handleEmailChange}
                    autoFocus
                    required
                />
                <div className={classes.textFieldWrap}>
                    <TextField
                        inputRef={inputField}
                        variant="outlined"
                        size="small"
                        label="Введите пароль"
                        className={classes.inputLine}
                        type='password'
                        value={password}
                        error={/^.{0,5}$/.test(password)}
                        onChange={handlePasswordChange}
                        autoFocus
                        required
                    />
                    <FormHelperText id="component-helper-text"
                        error={/^.{0,5}$/.test(password)}>
                        {/^.{0,5}$/.test(password) ? 'Длина пароля должна быть не менее 6 символов' : ` `}
                    </FormHelperText>
                </div>
                <div className={classes.regBtnWrap}>
                    {error && <p>{error}</p>}
                    <Button
                        size="large"
                        type="submit"
                        variant="contained"
                        endIcon={<AssignmentInd />}
                        className={classes.regBtn}
                    >
                        Регистрация
                    </Button>
                </div>

                <p>Уже зарегестрированы?
                    <Link to='/login' className={classes.loginLink}>
                        <Button
                            size="large"
                            type="submit"
                            variant="contained"
                            endIcon={<Login />}
                            className={classes.logBtn}
                        >
                            Войти
                        </Button>
                    </Link>
                </p>
            </fieldset>
        </form>
    </div>
}

export default Registration