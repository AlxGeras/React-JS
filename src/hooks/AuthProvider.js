import { async } from '@firebase/util';
import React, { useState } from 'react'
import { getAuth, onAuthStateChanged, signInWithEmailAndPassword, signOut } from 'firebase/auth'
import fireBaseConfig from '../services/firebaseConfig';

let AuthContext = React.createContext(null);

export function AuthProvider({ children }) {
    const [user, setUser] = useState();

    let signin = async (newUser, callback) => {
        const auth = getAuth(fireBaseConfig)
        await signInWithEmailAndPassword(auth, newUser.email, newUser.password)
        await onAuthStateChanged(auth, (user) => {
            if (user) {
                setUser(user)
            }
        })
        callback();
    }

    let singnout = async (callback) => {
        const auth = getAuth(fireBaseConfig)
        await signOut(auth)
        setUser(null);
        callback();
    }

    let value = { user, signin, singnout };

    return <AuthContext.Provider value={value}>{children}</AuthContext.Provider>
}

const useAuth = () => {
    return React.useContext(AuthContext)
}

export default useAuth;