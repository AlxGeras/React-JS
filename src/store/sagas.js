import {takeLatest, put, delay} from 'redux-saga/effects'
import { addMessage, ADD_MESSAGE_WITH_SAGA } from './messages/actions'

function* onAddMessageWithSga(action) {
    yield put(addMessage(action.payload.chatId, action.payload.message))
    if (action.payload.message.author !== 'bot'){
        const botMessage = {text: 'Ваше сообщение доставлено', author: 'bot'}
        yield delay(2000)
        yield put(addMessage(action.payload.chatId, botMessage) )
    }
}

function* mySaga() {
    yield takeLatest(ADD_MESSAGE_WITH_SAGA, onAddMessageWithSga);
}

export default mySaga