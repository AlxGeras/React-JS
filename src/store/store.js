import {createStore, combineReducers, applyMiddleware, compose} from 'redux';
import chatsReducer from './chats/reducer.js';
import messagesReducer from './messages/reducer.js';
import profileReducer from './profile/reducer.js';
import {persistStore, persistReducer} from 'redux-persist'
import storage from 'redux-persist/lib/storage'
import thunk from 'redux-thunk'
import gistsReducer from './gists/reducer.js';

const combineEnchancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose

const persistConfig = {
    key: 'root', 
    storage                             
}

const reducers = combineReducers({
    profile: profileReducer,
    chats: chatsReducer,
    messages: messagesReducer,
    gists: gistsReducer
})

const persistedReducer = persistReducer(persistConfig, reducers)

export const store = createStore(
    persistedReducer,
    combineEnchancers(applyMiddleware(thunk)));

 const persistor = persistStore(store);

 export default persistor

