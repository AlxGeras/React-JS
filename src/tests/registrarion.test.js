import { fireEvent, render, screen } from '@testing-library/react';
import Registration from "../components/Registration";
import { Provider } from "react-redux";
import { store } from '../store/store';
import { AuthProvider } from "../hooks/AuthProvider";
import { BrowserRouter } from "react-router-dom";

describe("Input tests", () => {
    it("test input screen handle function", () => {

        const view = render(
            <Provider store={store}>
                <AuthProvider>
                    <BrowserRouter>
                        <Registration />
                    </BrowserRouter>
                </AuthProvider>
            </Provider>

        );

        expect(view).toMatchSnapshot();


    });
});
