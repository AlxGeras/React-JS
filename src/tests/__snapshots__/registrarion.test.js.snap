// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input tests test input screen handle function 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="makeStyles-formWrap-8"
      >
        <div
          class="Toastify"
        />
        <form>
          <fieldset
            class="makeStyles-inputScreen-1"
          >
            <legend
              class="makeStyles-legend-7"
            >
              Форма регистрации
            </legend>
            <div
              class="MuiFormControl-root MuiTextField-root makeStyles-inputLine-2 css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="mui-1"
                id="mui-1-label"
              >
                Введите ваше имя
                <span
                  aria-hidden="true"
                  class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-describedby="mui-1-helper-text"
                  aria-invalid="false"
                  class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="mui-1"
                  required=""
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-1ftyaf0"
                  >
                    <span>
                      Введите ваше имя
                       
                      *
                    </span>
                  </legend>
                </fieldset>
              </div>
              <p
                class="MuiFormHelperText-root MuiFormHelperText-sizeSmall MuiFormHelperText-contained Mui-required css-k4qjio-MuiFormHelperText-root"
                id="mui-1-helper-text"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </p>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root makeStyles-inputLine-2 css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="mui-2"
                id="mui-2-label"
              >
                Введите email
                <span
                  aria-hidden="true"
                  class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-describedby="mui-2-helper-text"
                  aria-invalid="false"
                  class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="mui-2"
                  required=""
                  type="email"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-1ftyaf0"
                  >
                    <span>
                      Введите email
                       
                      *
                    </span>
                  </legend>
                </fieldset>
              </div>
              <p
                class="MuiFormHelperText-root MuiFormHelperText-sizeSmall MuiFormHelperText-contained Mui-required css-k4qjio-MuiFormHelperText-root"
                id="mui-2-helper-text"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </p>
            </div>
            <div
              class="makeStyles-textFieldWrap-6"
            >
              <div
                class="MuiFormControl-root MuiTextField-root makeStyles-inputLine-2 css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-error Mui-focused Mui-required css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="true"
                  for="mui-3"
                  id="mui-3-label"
                >
                  Введите пароль
                  <span
                    aria-hidden="true"
                    class="MuiInputLabel-asterisk MuiFormLabel-asterisk Mui-error css-wgai2y-MuiFormLabel-asterisk"
                  >
                     
                    *
                  </span>
                </label>
                <div
                  class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary Mui-error Mui-focused MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="true"
                    class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                    id="mui-3"
                    required=""
                    type="password"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-1in441m"
                    >
                      <span>
                        Введите пароль
                         
                        *
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <p
                class="MuiFormHelperText-root Mui-error css-1d1r5q-MuiFormHelperText-root"
                id="component-helper-text"
              >
                Длина пароля должна быть не менее 6 символов
              </p>
            </div>
            <div
              class="makeStyles-regBtnWrap-3"
            >
              
              <button
                class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root makeStyles-regBtn-4 css-jh47zj-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="submit"
              >
                Регистрация
                <span
                  class="MuiButton-endIcon MuiButton-iconSizeLarge css-z4rqyd-MuiButton-endIcon"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="AssignmentIndIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 3h-4.18C14.4 1.84 13.3 1 12 1c-1.3 0-2.4.84-2.82 2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-7 0c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm0 4c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm6 12H6v-1.4c0-2 4-3.1 6-3.1s6 1.1 6 3.1V19z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <p>
              Уже зарегестрированы?
              <a
                class="makeStyles-loginLink-9"
                href="/login"
              >
                <button
                  class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root makeStyles-logBtn-5 css-jh47zj-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="submit"
                >
                  Войти
                  <span
                    class="MuiButton-endIcon MuiButton-iconSizeLarge css-z4rqyd-MuiButton-endIcon"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="LoginIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M11 7 9.6 8.4l2.6 2.6H2v2h10.2l-2.6 2.6L11 17l5-5-5-5zm9 12h-8v2h8c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-8v2h8v14z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </a>
            </p>
          </fieldset>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="makeStyles-formWrap-8"
    >
      <div
        class="Toastify"
      />
      <form>
        <fieldset
          class="makeStyles-inputScreen-1"
        >
          <legend
            class="makeStyles-legend-7"
          >
            Форма регистрации
          </legend>
          <div
            class="MuiFormControl-root MuiTextField-root makeStyles-inputLine-2 css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="mui-1"
              id="mui-1-label"
            >
              Введите ваше имя
              <span
                aria-hidden="true"
                class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-describedby="mui-1-helper-text"
                aria-invalid="false"
                class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                id="mui-1"
                required=""
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1ftyaf0"
                >
                  <span>
                    Введите ваше имя
                     
                    *
                  </span>
                </legend>
              </fieldset>
            </div>
            <p
              class="MuiFormHelperText-root MuiFormHelperText-sizeSmall MuiFormHelperText-contained Mui-required css-k4qjio-MuiFormHelperText-root"
              id="mui-1-helper-text"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </p>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root makeStyles-inputLine-2 css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="mui-2"
              id="mui-2-label"
            >
              Введите email
              <span
                aria-hidden="true"
                class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-describedby="mui-2-helper-text"
                aria-invalid="false"
                class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                id="mui-2"
                required=""
                type="email"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1ftyaf0"
                >
                  <span>
                    Введите email
                     
                    *
                  </span>
                </legend>
              </fieldset>
            </div>
            <p
              class="MuiFormHelperText-root MuiFormHelperText-sizeSmall MuiFormHelperText-contained Mui-required css-k4qjio-MuiFormHelperText-root"
              id="mui-2-helper-text"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </p>
          </div>
          <div
            class="makeStyles-textFieldWrap-6"
          >
            <div
              class="MuiFormControl-root MuiTextField-root makeStyles-inputLine-2 css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-error Mui-focused Mui-required css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="true"
                for="mui-3"
                id="mui-3-label"
              >
                Введите пароль
                <span
                  aria-hidden="true"
                  class="MuiInputLabel-asterisk MuiFormLabel-asterisk Mui-error css-wgai2y-MuiFormLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary Mui-error Mui-focused MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="true"
                  class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="mui-3"
                  required=""
                  type="password"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-1in441m"
                  >
                    <span>
                      Введите пароль
                       
                      *
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <p
              class="MuiFormHelperText-root Mui-error css-1d1r5q-MuiFormHelperText-root"
              id="component-helper-text"
            >
              Длина пароля должна быть не менее 6 символов
            </p>
          </div>
          <div
            class="makeStyles-regBtnWrap-3"
          >
            
            <button
              class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root makeStyles-regBtn-4 css-jh47zj-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="submit"
            >
              Регистрация
              <span
                class="MuiButton-endIcon MuiButton-iconSizeLarge css-z4rqyd-MuiButton-endIcon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="AssignmentIndIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 3h-4.18C14.4 1.84 13.3 1 12 1c-1.3 0-2.4.84-2.82 2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-7 0c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm0 4c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm6 12H6v-1.4c0-2 4-3.1 6-3.1s6 1.1 6 3.1V19z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <p>
            Уже зарегестрированы?
            <a
              class="makeStyles-loginLink-9"
              href="/login"
            >
              <button
                class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root makeStyles-logBtn-5 css-jh47zj-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="submit"
              >
                Войти
                <span
                  class="MuiButton-endIcon MuiButton-iconSizeLarge css-z4rqyd-MuiButton-endIcon"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="LoginIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M11 7 9.6 8.4l2.6 2.6H2v2h10.2l-2.6 2.6L11 17l5-5-5-5zm9 12h-8v2h8c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-8v2h8v14z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </a>
          </p>
        </fieldset>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
